tracing-core/src/dispatcher.rs

```diff
@@ -273,6 +290,10 @@ pub fn set_global_default(dispatcher: Dispatch) -> Result<(), SetGlobalDefaultEr
     {
         unsafe {
             GLOBAL_DISPATCH = Some(dispatcher);
+            GLOBAL_DEFAULT_DISPATCH.store(
+                GLOBAL_DISPATCH.as_mut().unwrap() as *mut _,
+                Ordering::SeqCst,
+            );
```

Comment:
ideally, i think this would be a `compare_exchange` so that it fails if the global dispatch value was something other than the initial default `NO_DISPATCH` value. we could panic if the CAS fails (at least in debug mode)